common/compositor/va/varenderer.cpp

```diff
@@ -440,18 +448,33 @@ bool VARenderer::UpdateCaps() {
       if (use_default) {
         itr->second.value_ = itr->second.caps_.range.default_value;
       }
-      cbparam.type = VAProcFilterColorBalance;
-      cbparam.value = itr->second.value_;
-      cbparam.attrib = itr->second.caps_.type;
-      if (!cb_elements[static_cast<int>(itr->first)].CreateBuffer(
-              va_context_, VAProcFilterParameterBufferType,
-              sizeof(VAProcFilterParameterBufferColorBalance), 1, &cbparam)) {
-        return false;
-      }
-      filters_.push_back(cb_elements[static_cast<int>(itr->first)].buffer());
+      cbparam[index].type = VAProcFilterColorBalance;
+      cbparam[index].value = itr->second.value_;
+      cbparam[index].attrib = itr->second.caps_.type;
+      index++;
     }
   }
 
+  // workarround if filter num is 0, then set HSBC/Deinterlace will meet green
+  // screen.
+  if (!index) {
```

Comment:
@feiwan1 can you help me understand the work around needed here? I'm having a hard time understanding how this prevents the issue. Thanks!